threatingestor/sources/github.py

```diff
@@ -8,29 +8,56 @@
 SEARCH_URL = "https://api.github.com/search/repositories"
 
 class Plugin(Source):
-
-    def __init__(self, name, search):
+    """Github Source Plugin"""
+    def __init__(self, name, search, username="", token=""):
         self.name = name
         self.search = search
 
+        if username and token:
+            self.auth = (username, token)
+        else:
+            self.auth = None
+
+    def make_requests(self, url, params, auth):
+        """Returns a list of artifacts"""
+        # Iterates through pages of results from query
+        artifact_list = []
```

Comment:
I think we could improve the separation of concerns here by having this method return a list of repository results rather than a list of artifacts, and then loop over those results in `run` to create the Task artifacts.